shared/src/sources/balancer/pool_fetching.rs

```diff
@@ -105,19 +106,42 @@ impl BalancerPoolEvaluating for WeightedPool {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct AmplificationParameter {
+    factor: U256,
+    precision: U256,
+}
+
+impl AmplificationParameter {
+    pub fn new(factor: U256, precision: U256) -> Self {
+        Self { factor, precision }
+    }
+
+    /// This is the format used to pass into smart contracts.
+    pub fn as_u256(&self) -> Option<U256> {
+        self.factor.checked_mul(self.precision)
+    }
+
+    /// This is the format used to pass along to HTTP solver.
+    pub fn as_big_rational(&self) -> BigRational {
+        BigRational::new(self.factor.to_big_int(), self.precision.to_big_int())
```

Comment:
@nlordell do you have any suggestions on how best to handle the completely unexpected scenario where the precision is 0?